llm/llama.go

```diff
@@ -558,6 +558,19 @@ func (llm *llama) Predict(ctx context.Context, prevContext []int, prompt string,
 		"penalize_nl":       llm.PenalizeNewline,
 		"seed":              llm.Seed,
 		"stop":              llm.Stop,
+		"grammar":           llm.Grammar,
+	}
+
+	// If json schema is provided, convert it to a grammar
+	if llm.Schema != "" {
+		if llm.Grammar != "" {
+			return fmt.Errorf("cannot use both a grammar and a json schema, please provide only one")
+		}
+		grammar, err := format.SchemaToGrammar(llm.Schema, nil)
```

Comment:
@Lwrless It looks like the latest changes from the `main` branch break this line, as a variable called `format` is introduced for the JSON mode (https://github.com/jmorganca/ollama/pull/1051).

I locally resolved this by renaming the argument to `outputFormat`, and also moving the whole conditional down (so that a schema supersedes generic JSON mode).

Reply:
Thank you for pointing out the issue! I apologize for not testing after merging `main`.

The mistake is amended. And now to use JSON schema, the request's `format` option should be set to `json`.

Reply:
Thanks for implementing this feature! In the short time testing it yesterday I have been having a great time using it to reduce halucinations for function calling! :)